src/SFML/Graphics/Texture.cpp

```diff
@@ -400,6 +401,22 @@ Image Texture::copyToImage() const
         glCheck(GLEXT_glDeleteFramebuffers(1, &frameBuffer));
 
         glCheck(GLEXT_glBindFramebuffer(GLEXT_GL_FRAMEBUFFER, static_cast<GLuint>(previousFrameBuffer)));
+
+        if (m_pixelsFlipped)
+        {
+            // Flip the texture vertically
+            const auto stride             = static_cast<std::ptrdiff_t>(m_size.x * 4);
+            auto       currentRowIterator = pixels.begin();
+            auto       nextRowIterator    = pixels.begin() + stride;
+            auto       reverseRowIterator = pixels.begin() + (stride * static_cast<std::ptrdiff_t>(m_size.y - 1));
```

Comment:
Maybe we can use [`std::vector::rbegin()`](https://en.cppreference.com/w/cpp/container/vector/rbegin) here?